### PR TITLE
Adding simple support for S3 backed repositories

### PR DESCRIPTION
**Pull request by ljakimczuk**
Hi! I'd like to add support for the S3-backed repositories.

Well, if this is not the best way of doing this, please share with me any suggestions and I would gladly put fixes. My way of thinking was simple, at the beginning I thought of separating the `repo_index` and the `from_repo` into something like `[ s3 | http ]_repo_index` and `from_[ s3 | http ]_repo` in order to somehow highlight the difference between the two back-ends. Then, I imagined from the user's perspective it may be smoother to be allowed to stick to the `repo_index` and the `from_repo` and then just feed them with the new URL, instead of being introduced with new interfaces. In result, I move the fetching of index/Charts into the `_get` functions.

One more think to mention, at least the Chartmuseum's `index.yaml` may only have the relative URLs instead of absolute ones. This is one of the two reasons for the extra `if` and `join`.

### CODE REVIEW

File: pyhelm/repo.py [hunk 1]
```diff
@@ -2,23 +2,82 @@
...
+        )
+
+        return file_object['Body'].read()
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchBucket':
+            raise RuntimeError('%s repository not found' % file_url_parsed.netloc)
+        elif e.response['Error']['Code'] == 'NoSuchKey':
+            raise RuntimeError('%s not found in the repository' % file_url_parsed.path.strip('/'))
+        else:
+            pass
```
Comment (yanivoliver) [inline pyhelm/repo.py]: What would you expect the behaviour to be in this case? I'd re-raise the exception

File: pyhelm/repo.py [hunk 2]
```diff
@@ -29,14 +88,28 @@ def from_repo(repo_url, chart, version=None):
...
                                      versions)
-
     try:
         metadata = sorted(versions, key=lambda x: x['version'])[0]
         for url in metadata['urls']:
             fname = url.split('/')[-1]
             try:
-                req = requests.get(url, stream=True)
-                fobj = cStringIO.StringIO(req.content)
+                if repo_scheme == 's3':
```
Comment (yanivoliver) [inline pyhelm/repo.py]: This code block is very similar to the one in _repo_index_. Perhaps create a helper method that will contain the common functionality, and have this method and _repo_index_ only wrap it accordingly (e.g. wrap with yaml.load)

File: requirements.txt [hunk 3]
```diff
@@ -4,4 +4,6 @@ grpcio-tools==1.1.3
 protobuf==3.2.0
 supermutes==0.2.5
 requests==2.14.2
-PyYAML==3.12
\ No newline at end of file
+PyYAML==3.12
```
Comment (yanivoliver) [inline requirements.txt]: Another PR removed version pinning from our requirements. Please remove yours and resolve the conflict

**Comment by yanivoliver** [comment]
@ljakimczuk - Sorry for taking so long to over your PR. Thanks again!

File: pyhelm/repo.py (hunk 1 above)
Comment (ljakimczuk) [reply pyhelm/repo.py]: I would actually expect the code to stop, so I totally agree with re-raising the exception. To be honest I don't remember why I placed the `pass` there, probably it's due to being sloppy..

**Comment by yanivoliver** [comment]
Looks great. Thanks for the fixes

**Comment by ljakimczuk** [comment]
@yanivoliver Thanks!

**Pull request merged**